python/dgl/nn/pytorch/conv/chebconv.py

```diff
@@ -86,42 +92,52 @@ def forward(self, graph, feat, lambda_max=None):
-            norm = th.pow(
-                graph.in_degrees().float().clamp(min=1), -0.5).unsqueeze(-1).to(feat.device)
+            D_sqrt = th.pow(graph.in_degrees().float().clamp(
+                min=1), -0.5).unsqueeze(-1).to(feat.device)
+
             if lambda_max is None:
-                lambda_max = laplacian_lambda_max(graph)
+                try:
+                    lambda_max = dgl.laplacian_lambda_max(graph)
+                except BaseException:
+                    # if the largest eigonvalue is not found
+                    lambda_max = th.Tensor(2).to(feat.device)
+
             if isinstance(lambda_max, list):
                 lambda_max = th.Tensor(lambda_max).to(feat.device)
             if lambda_max.dim() == 1:
                 lambda_max = lambda_max.unsqueeze(-1)  # (B,) to (B, 1)
+
             # broadcast from (B, 1) to (N, 1)
-            lambda_max = broadcast_nodes(graph, lambda_max)
-            # T0(X)
-            Tx_0 = feat
-            rst = self.fc[0](Tx_0)
-            # T1(X)
+            lambda_max = dgl.broadcast_nodes(graph, lambda_max)
+
+            # X_0(f)
+            Xt = X_0 = signal
+
+            # X_1(f)
             if self._k > 1:
-                graph.ndata['h'] = Tx_0 * norm
-                graph.update_all(fn.copy_u('h', 'm'), fn.sum('m', 'h'))
-                h = graph.ndata.pop('h') * norm
-                # Λ = 2 * (I - D ^ -1/2 A D ^ -1/2) / lambda_max - I
-                #   = - 2(D ^ -1/2 A D ^ -1/2) / lambda_max + (2 / lambda_max - 1) I
-                Tx_1 = -2. * h / lambda_max + Tx_0 * (2. / lambda_max - 1)
-                rst = rst + self.fc[1](Tx_1)
-            # Ti(x), i = 2...k
+                re_norm = 2. / lambda_max
+                h = unnLaplacian(X_0, D_sqrt, graph)
+
+                X_1 = - re_norm * h + X_0 * (re_norm - 1)
+                Xt = torch.cat((Xt, X_1), 1)
+
+            # Xi(x), i = 2...k
             for i in range(2, self._k):
-                graph.ndata['h'] = Tx_1 * norm
-                graph.update_all(fn.copy_u('h', 'm'), fn.sum('m', 'h'))
-                h = graph.ndata.pop('h') * norm
-                # Tx_k = 2 * Λ * Tx_(k-1) - Tx_(k-2)
-                #      = - 4(D ^ -1/2 A D ^ -1/2) / lambda_max Tx_(k-1) +
-                #        (4 / lambda_max - 2) Tx_(k-1) -
-                #        Tx_(k-2)
-                Tx_2 = -4. * h / lambda_max + Tx_1 * (4. / lambda_max - 2) - Tx_0
-                rst = rst + self.fc[i](Tx_2)
-                Tx_1, Tx_0 = Tx_2, Tx_1
-            # add bias
-            if self.bias is not None:
-                rst = rst + self.bias
-            return rst
+                h = unnLaplacian(X_1, D_sqrt, graph)
+                X_i = - 2 * re_norm * h + X_1 * 2 * (re_norm - 1) - X_0
+
+                Xt = torch.cat((Xt, X_i), 1)
+                X_1, X_0 = X_i, X_1
+
+            # forward pass
+            graph.ndata['h'] = Xt
+            graph.apply_nodes(func=self.apply_mod)
```

Comment:
It seems you are concatenating X_0, X_1, ... X_t and apply a linear transformation on the concatenation, however, this is not equivalent to the original paper where the weight W^k corresponding to X_k would not influence X_j(j≠k).

Reply:
Hi, I’m not sure what you mean by ‘influence’, could you please rephrase? 
In this implementation there is first a concatenation of all the polynomes making the feature at each node ( Fin*k ) dimensional. After, a linear layer with ReLu of size (Fin*k -> Fout ) is applied. As the linear layer has as one parameter for each dimension of Xt I don’t think that qualifies as influencing each other (?).

Reply:
Sorry I misunderstood your code, you are right.